pkg/cmd/dcos_auth_listproviders.go

```diff
@@ -0,0 +1,120 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+// These are the different auth types that DC/OS supports with the names that they'll be given from the providers
+// endpoint.
+const (
+	AuthTypeDCOSUidPassword = "dcos-uid-password"
+	AuthTypeDCOSUidServiceKey = "dcos-uid-servicekey"
+	AuthTypeDCOSUidPasswordLDAP = "dcos-uid-password-ldap"
+	AuthTypeSAMLSpInitiated = "saml-sp-initiated"
+	AuthTypeOIDCAuthCodeFlow = "oidc-authorization-code-flow"
+	AuthTypeOIDCImplicitFlow = "oidc-implicit-flow"
+)
+
+var jsonOutput bool
+
+// authCmd represents the `dcos auth` subcommand.
+var authListProvidersCmd = &cobra.Command{
+	Use: "list-providers",
+	Run: listProviders,
+}
+
+func init() {
+	authCmd.AddCommand(authListProvidersCmd)
+	authListProvidersCmd.Flags().BoolVar(&jsonOutput, "json", false,
+		"returns providers in json format")
+}
+
+func listProviders(cmd *cobra.Command, args []string) {
+	providers, err := getProviders()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if jsonOutput {
+		// re-marshal it into json with indents added in for pretty printing
+		out, err := json.MarshalIndent(providers, "", "\t")
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		fmt.Println(string(out))
+	} else {
+		table := tablewriter.NewWriter(os.Stdout)
+		table.SetHeader([]string{"PROVIDER ID", "AUTHENTICATION TYPE"})
+		// turn off wrapping because it seems to wrap even if the column is set to be wide enough
+		table.SetAutoWrapText(false)
+		table.SetBorder(false)
+		table.SetRowSeparator(" ")
+		table.SetColumnSeparator(" ")
+		table.SetCenterSeparator(" ")
+
+		for name, provider := range *providers {
+			desc, err := authTypeDescription(provider.AuthenticationType, provider)
+			if err != nil {
+				fmt.Printf("Unknown authentication type %s", provider.AuthenticationType)
+				os.Exit(1)
+			}
+			table.Append([]string{name, desc})
+		}
+		table.Render()
+	}
+}
+
+func getProviders() (*map[string]authProvider, error) {
+	var config = dcosConfig.Config
+	var client = httpclient.New(config)
+	var response, err = client.Get("/acs/api/v1/auth/providers")
```

Comment:
the short variable declaration is more common in go, that's what we currently use, eg. `response, err := client.Get("/acs/api/v1/auth/providers")`